tests/check/check_test.go

```diff
@@ -126,3 +132,64 @@ func runTests(t *testing.T, client pb.OpenFGAServiceClient, tests checkTests) {
 		})
 	}
 }
+
+func testBadAuthModelID(t *testing.T, engine string) {
+
+	cfg := run.MustDefaultConfigWithRandomPorts()
+	cfg.Log.Level = "none"
+	cfg.Datastore.Engine = engine
+
+	cancel := tests.StartServer(t, cfg)
+	defer cancel()
+
+	conn := tests.Connect(t, cfg.GRPC.Addr)
+	defer conn.Close()
+	client := pb.NewOpenFGAServiceClient(conn)
+
+	ctx := context.Background()
+	resp, err := client.CreateStore(ctx, &pb.CreateStoreRequest{Name: "bad auth id"})
+	require.NoError(t, err)
+
+	storeID := resp.GetId()
+	_, err = client.WriteAuthorizationModel(ctx, &pb.WriteAuthorizationModelRequest{
+		StoreId:       storeID,
+		SchemaVersion: typesystem.SchemaVersion1_1,
+		TypeDefinitions: []*pb.TypeDefinition{
+			{
```

Comment:
Please use the DSL parser and a DSL string here instead of an inline TypeDefinition struct.

e.g.

```
type user

type doc
  relations
    define viewer: [user] as self
    define can_view as viewer
```
and `parser.MustParse(modelStr)`

Reply:
done